prog/weaver/main.go

```diff
@@ -452,18 +452,21 @@ func main() {
 		go expose(allocator, defaultSubnet, bridgeConfig.WeaveBridgeName, bridgeConfig.AWSVPC)
 	}
 
-	// Start Weave Proxy:
-	proxy, err := weaveproxy.NewProxy(*proxyConfig)
-	if err != nil {
-		Log.Fatalf("Could not start proxy: %s", err)
+	if proxyConfig.Disabled {
+		signals.SignalHandlerLoop(common.Log, router)
+	} else {
+		// Start Weave Proxy:
+		proxy, err := weaveproxy.NewProxy(*proxyConfig)
+		if err != nil {
+			Log.Fatalf("Could not start proxy: %s", err)
+		}
+		defer proxy.Stop()
+		listeners := proxy.Listen()
+		proxy.AttachExistingContainers()
+		go proxy.Serve(listeners)
+		go proxy.ListenAndServeStatus("/home/weave/status.sock")
+		signals.SignalHandlerLoop(common.Log, proxy, router)
```

Comment:
Having thought about it a little more, you already have `defer proxy.Stop()` so it's pointless passing proxy in here.
And there is no `plugin.Stop` so that's a separate issue.